scattering/scattering.py

```diff
@@ -72,29 +94,48 @@ def structure_factor(
     L = np.min(trj.unitcell_lengths)
 
     top = trj.topology
-    elements = set([a.element for a in top.atoms])
+    unique_residues = []
+
+    for a in top.residues:
+        if a.name not in unique_residues:
+            unique_residues.append(a.name)
+            residue_atoms = []
+            for b in a.atoms:
+                residue_atoms.append(b.name)
+            print(
+                "The residue name is {} and it contains {}".format(
+                    a.name, residue_atoms
+                )
+            )
 
+    elements = set([a.element for a in top.atoms])
+    names = set([a.name for a in top.atoms])
     compositions = dict()
     sq = dict()
 
     Q = np.logspace(np.log10(Q_range[0]), np.log10(Q_range[1]), num=n_points)
     S = np.zeros(shape=(len(Q)))
 
-    for elem in elements:
-        compositions[elem.symbol] = (
-            len(top.select("element {}".format(elem.symbol))) / trj.n_atoms
-        )
+    for name in names:
+        compositions[name] = len(top.select("name {}".format(name))) / trj.n_atoms
 
     # Compute partial structure factors
     print("Computing structure factors ...")
-    for (elem1, elem2) in it.product(elements, repeat=2):
-        e1 = elem1.symbol
-        e2 = elem2.symbol
 
-        sq["{0}{1}".format(e1, e2)] = partial_structure_factor(
+    atoms = set()
+    for name in names:
+        element = find_element(name, top)
```

Comment:
I want to avoid adding another `Class` if possible, especially since the `md.Trajectory` should carry all necessary information to carry out this analysis.  What if we did something like this instead:

```
114     atom_select = list()
115     for name in names:
116         atom_select.append(temp_trj.top.select(f"name {name}")[0])
120 
121     name_trj = temp_trj.atom_slice(atom_select)
122     atoms = [atom for atom in name_trj.top.atoms]
```